src/devices/directv.c

```diff
@@ -0,0 +1,532 @@
+    [0x62] = "2 REPEAT",
+    [0x63] = "3 REPEAT",
+    [0x64] = "4 REPEAT",
+    [0x65] = "5 REPEAT",
+    [0x66] = "6 REPEAT",
+    [0x67] = "7 REPEAT",
+    [0x68] = "8 REPEAT",
+    [0x69] = "9 REPEAT",
+    [0x73] = "FORMAT",
+    [0x75] = "FORMAT REPEAT",
+    [0x80] = "DTV: DTV&TV POWER ON",
+    [0x81] = "DTV: DTV&TV POWER OFF",
+    [0xD6] = "SELECT RELEASE"
+};
+
+const uint8_t *get_dtv_button_label(uint8_t button_id)
+{
+    const uint8_t *label = dtv_button_label[button_id];
+    if (!label) {
+        label = dtv_button_label[0];
+    }
+    return label;
+}
+
+
+static inline int bit(const uint8_t *bytes, unsigned bit)
+{
+    return bytes[bit >> 3] >> (7 - (bit & 7)) & 1;
+}
+
+// similar to bitbuffer_search in bitbuffer.h, but allows us to search a custom bitrow_t
+unsigned bitrow_search(bitrow_t const bitrow, unsigned bit_len, unsigned start,
+        const uint8_t *pattern, unsigned pattern_bits_len)
+{
+    unsigned ipos = start;
+    unsigned ppos = 0; // cursor on init pattern
+
+    while (ipos < bit_len && ppos < pattern_bits_len) {
+        if (bit(bitrow, ipos) == bit(pattern, ppos)) {
+            ppos++;
+            ipos++;
+            if (ppos == pattern_bits_len)
+                return ipos - pattern_bits_len;
+        }
+        else {
+            ipos -= ppos;
+            ipos++;
+            ppos = 0;
+        }
+    }
+
+    // Not found
+    return bit_len;
+}
+
+/// Set a single bit in a bitrow at bit_idx position.  Assume success, no bounds checking, so be careful!
+void bitrow_set_bit(bitrow_t bitrow, unsigned bit_idx, unsigned bit_val)
+{
+    uint8_t bit_mask;
+    if (bit_val == 0) {
+        bitrow[bit_idx >> 3] &= ~(1 << (7 - (bit_idx & 7)));
+    } 
+    else {
+        bitrow[bit_idx >> 3] |= (1 << (7 - (bit_idx & 7)));
+    }
+}
+
+static int directv_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+
+    data_t *data;
+    int r;                   // a row index
+    uint8_t bitrow[15];      // space for a possibly modified bitbuffer row
+    uint8_t bit_len;         // row length is variable, so need to keep track of this
+    uint8_t dtv[5] = {0};    // enough for exactly 40 bits of decoded data
+  
+    // Signal is reset by rtl_433 before recognizing rows, so in practice, there's only one row.
+    // It would be useful to catch rows in signal so that there'd only be one decoded row per
+    // signal and we could count repeats to report the length of the signal, but that's not
+    // supported yet.  It seems the gap between rows exceeds the ook_hysteresis threshold in
+    // pulse_detect.c:  int16_t const ook_hysteresis = ook_threshold / 8; // ±12%
+    // and changing this value isn't the right direction, nor does this even work for this signal.
+    // Grouping rows in a signal like this will need to be supported in some other way, and this 
+    // support is not yet available in rtl_433.
+    // For now, we'll decode the signal in the bitbuffer assuming it is only one row.
```

Comment:
Proper grouping and segmentation of rows are an issue in other signals as well. They break the heuristics that is used. And when we go for FSK signals they are a magnitude shorter then the OOK ones. All in all the grouping properties doesn't apply. It's hard to decode all kinds of random signals.